Automattic/jetpack
# Limit post meta_values that gets Synced to 2MB

meta values can have content of any length. We've seen that some sites create post_meta that exceeds 10MB at a time which can lead to processing errors and Sync no longer functional on their site as the action continually tries to be synced and processed.

This PR will truncate post meta values that exceeds 2MB so that we can still sync the action but the meta_value will be empty.

#### Changes proposed in this Pull Request:
* Limit meta_value that is syncable to 2MB.

Does this pull request change what data or activity we track or use?
No
Testing instructions:
Unit Tests added to test the imposed limit.

Manual Testing

1. Test site needs to have an active jetpack connection
2. Create a new post
3. add meta that exceeds 2000000 characters in length `update_post_meta`
4. From WP.com sandbox run get_post_meta for the meta you created. Verify it returns an empty string.
5. update the meta value so it is less than 2000000 characters in length
6. From WP.com sandbox run get_post_meta for the meta you created. Verify it returns the value from the jetpack site.
7. Verify that objects/arrays still save properly for meta_values

Verify sync/object also adheres to the limit.
1. access wp shell
2. add meta that exceeds 2000000 characters in length `update_post_meta`
3. $module = Automattic/Jetpack/Sync/Modules::get( 'meta' );
4. $meta = $module->get_object_by_id( 'post', POST_ID, META_KEY );
5. Verify $meta['meta_value'] is an empty string.
6. add meta that is less than 2000000 characters in length `update_post_meta`
7. $meta = $module->get_object_by_id( 'post', POST_ID, META_KEY );
8. Verify $meta['meta_value'] is the value you added/updated
9. Verify that adding/updating of objects/arrays to meta still functions

#### Proposed changelog entry for your changes:
* Jetpack_Sync :: Impose a max limit of 2MB on post meta values that are synced.
